sdk/core/Azure.Core/api/Azure.Core.net461.cs

```diff
@@ -1024,8 +1023,8 @@ public abstract partial class RetryPolicy : Azure.Core.Pipeline.HttpPipelinePoli
         protected internal virtual System.Threading.Tasks.ValueTask OnSendingRequestAsync(Azure.Core.HttpMessage message) { throw null; }
         public override void Process(Azure.Core.HttpMessage message, System.ReadOnlyMemory<Azure.Core.Pipeline.HttpPipelinePolicy> pipeline) { }
         public override System.Threading.Tasks.ValueTask ProcessAsync(Azure.Core.HttpMessage message, System.ReadOnlyMemory<Azure.Core.Pipeline.HttpPipelinePolicy> pipeline) { throw null; }
-        protected internal virtual bool ShouldRetry(Azure.Core.HttpMessage message) { throw null; }
-        protected internal virtual System.Threading.Tasks.ValueTask<bool> ShouldRetryAsync(Azure.Core.HttpMessage message) { throw null; }
+        protected internal virtual bool ShouldRetry(Azure.Core.HttpMessage message, System.Exception? exception) { throw null; }
```

Comment:
Would it make sense to provide a default for `exception`?

Reply:
I think it is probably better for it to be a required parameter. If customers override ShouldRetry and call into the base method, we wouldn't want them to forego passing the exception.